packages/odie-client/src/components/message/user-message.tsx

```diff
@@ -45,7 +47,7 @@ export const UserMessage = ( {
 					a: CustomALink,
 				} }
 			>
-				{ isRequestingHumanSupport ? forwardMessage : message.content }
+				{ isRequestingHumanSupport ? displayMessage : message.content }
```

Comment:
```suggestion
				{ isRequestingHumanSupport || hasCannedResponse ? displayMessage : message.content }
```

```
	const hasCannedResponse = isUserEligible && message.context?.flags?.canned_response;
```

I think this would make it easier to understand IMO. Your original method would be 2 variables for the conditional wording and 2 variables for the translated text itself. Then you could add the `isUserEligible` check to the `hasCannedResponse` check above. I feel like like this consolidates some of the logic.

Reply:
I agree that this proposal is easier to understand.

Reply:
I might be missing something, due to lack of coffee in my morning... But the code you are proposing is not equivalent to the existing one...

I think by amending your changes we are missing to print the canned_response, as message.content contains the canned_response, this means that message.content will only be used when both isRequestingHumanSupport and hasCannedResponse are false, which is a significant change.

Reply:
I see your point Daniel. I think what makes your solution complicated is the conditional within a conditional. Basically the fact that `displayMessage` has a conditional in it means that the logic is more like a `if elseif else` clause. Maybe your solution is the best though.

Reply:
Ahh you are absolutely correct I misread that. We would need to flip it to be something like:

```suggestion
				{ ! isRequestingHumanSupport || hasCannedResponse ? message.content : forwardMessage }
```

If we are requesting support show the forward message otherwise return message.
OR
If we have a canned response show message content otherwise return the forward message.